### PR TITLE
SSR Experimental

### PR DESCRIPTION
**Pull request by shashkovdanil**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context

- [ ] Closes
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [x] I've tested it at </ksm/collection>
- [x] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer?target=16SpvdDgKNiQ3c53DxX7refnQcKUD3uRNim3Z1HBJLNGrtSP&usdamount=0)

#### Community participation

- [x] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at a264c75</samp>

The pull request improves the compatibility, performance, and readability of various components by using custom hooks and services to access browser APIs and environment variables. It also refactors some code to simplify logic and avoid reactivity issues. The main files affected are `components/blog/BlogPost.vue`, `components/carousel/utils/useCarouselEvents.ts`, `components/chart/PriceChart.vue`, `components/collection/HeroButtons.vue`, `components/collection/unlockable/CountdownTimer.vue`, `components/collection/unlockable/UnlockableLoader.vue`, `components/common/ConnectWallet/ConnectWalletModal.vue`, `components/CookieBanner.vue`, `components/gallery/GalleryItemTabsPanel/GalleryItemTabsPanel.vue`, `components/MessageNotify.vue`, `components/Navbar.vue`, `components/profile/ProfileDetail.vue`, `components/base/MediaItem.vue`, `components/collection/unlockable/UnlockableHeroButtons.vue`, `components/CustomSubstackEmbed.vue`, `components/gallery/GalleryItemButton/GalleryItemShareBtn.vue`, `components/landing/MobileHeroBanner.vue`, `components/navbar/NotificationBoxButton.vue`, `components/navbar/ShoppingCartButton.vue`, and `components/search/SearchBar.vue`.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at a264c75</samp>

> _We're the masters of the browser, we don't need the `window`_
> _We use custom hooks and services, to make our code more nimble_
> _We check the server and the client, we avoid the errors of the `document`_
> _We're the masters of the browser, we rock the `vue` component_

### CODE REVIEW

**Comment by preschian** [comment]
> cc @yangwao you'll be glad

nice, but I can't open the preview page

Screenshot 2023-12-02 at 6 52 15 PM


need to test it on the Cloudflare pages also

**Review by roiLeo — CHANGES_REQUESTED** [review]
- Some of main components need ClientOnly (like darkmode)
- Dropdowns looks weird
- wallet doesn't work on reload

same thing as #7725

**Comment by roiLeo** [comment]
>     * Some of main components need ClientOnly (like darkmode)

Screenshot 2023-12-02 at 14-56-34 KodaDot - NFT Market Explorer

 
>     * Dropdowns looks weird

Screenshot 2023-12-02 at 14-55-54 KodaDot - NFT Market Explorer

 
> I fixed stores which connected with localStorage, it should be persist

✅

**Comment by yangwao** [comment]
> With latest fixes:

What I get on Incognito

image

**Comment by yangwao** [comment]
> is it mobile?

Mobile is usually even worse

image

> anyway 11 points it's really good result

appreciate all that work but I think we can find more asymmetric gains for least effort higher impact 

image

I think all that performance on landing we should rehaul landing as it is. Also, I sense we are loading assets below fold. and on strips those which are invisible to user. User usually won't use them right away tho, maybe lazy load them those which are not seen?

File: components/landing/SearchLanding.vue [hunk 1]
```diff
@@ -6,7 +6,10 @@
       width="740"
       height="1000"
       alt="" />
-    <img :src="landingImage[0]" class="landing-shapes" alt="" />
+    <img
+      src="/landing-shape-header-left-light.svg"
```
Comment (roiLeo) [inline components/landing/SearchLanding.vue:10]: oh this is new

File: assets/styles/components/_search.scss [hunk 2]
```diff
@@ -44,6 +44,10 @@
     z-index: 1;
     overflow: hidden;
     pointer-events: none;
+
+    @include ktheme() {
+      filter: theme('landing-shape-header');
+    }
```
Comment (roiLeo) [inline assets/styles/components/_search.scss:50]: why put code in this component?
https://caniuse.com/css-filters global 97% ok

File: components/CustomSubstackEmbed.vue [hunk 3]
```diff
@@ -3,27 +3,37 @@
...
+    substackUrl: 'kodadot.substack.com',
+    placeholder: 'jane.doe@kodadot.xyz',
+    theme: 'custom',
+    colors: {
+      primary: '#FF7AC3',
+      input: '#FFFFFF',
+      email: '#000000',
+      text: '#000000',
+    },
+  } as const
```
Comment (roiLeo) [inline components/CustomSubstackEmbed.vue:23]: as const?

File: plugins/polkadot.ts [hunk 4]
```diff
@@ -7,7 +6,13 @@ import { useChainStore } from '@/stores/chain'
 
 // }
 
-export default defineNuxtPlugin(() => {
+export default defineNuxtPlugin(async () => {
+  if (!process.client) {
```
Comment (roiLeo) [inline plugins/polkadot.ts:10]: rename file to `polkadot.client.ts` for only client side

File: plugins/pwa.ts [hunk 5]
```diff
@@ -5,6 +5,10 @@ import { clearSession } from '@/utils/cachingStrategy'
 
 // export default async () => {
 export default defineNuxtPlugin(async () => {
+  if (!process.client) {
+    return
+  }
```
Comment (roiLeo) [inline plugins/pwa.ts:10]: `pwa.client.ts`

**Review by roiLeo — COMMENTED** [review]
wow, I can see that this branch start to look good there is a lot of progress compared to #7725 

Now we have to test that nothing breaks: on first load, with cache, without cache, with reload etc..

**Pull request closed without merging**